### ISSUE TITLE
Dockerization

### ISSUE THREAD

**Issue by dmuth** [schollz/hostyoself#10]
I'd like to propose a Dockerized version of hostyoself so that the CLI version can be run via a `docker run` command.

I'll be happy to contribute a `Dockerfile` and push the Docker image to Docker Hub.  Just say the word! :-)

**Comment by schollz** [comment]
It was too easy, so I pushed it! Thanks though!

**Comment by dmuth** [comment]
FYI, the image doesn't seem to be uploaded to Docker Hub...?

```
$ docker pull schollz/hostyoself                                                                                                                                                     
Using default tag: latest
Error response from daemon: manifest for schollz/hostyoself:latest not found
```

**Comment by schollz** [comment]
Yeah, I forgot to use go generate, try now!

**Comment by dmuth** [comment]
Looks good!  I actually was planning on running hostyoself in relay mode on one of my severs, and since your build runs `host` by default, I ended up expanding on it in my fork.  I also fixed an issue with `COPY . .` which would cause the entire app to be recompiled if anything at all changed, such as the Dockerfile itself, or a Vim `.swp` file.

I'm going to send a PR shortly with what I did, but think of it less as a PR and more as a "pull this if you like".  If you would prefer to pick and choose from what I did in my commit and integrate it by hand, feel free to do so, and treat it was a contribution under the existing license for this app. :-)

-- Doug
